x/axelarnet/types/evm_translator.go

```diff
@@ -228,14 +237,26 @@ func buildArguments(argTypes []string) (abi.Arguments, error) {
 }
 
 func checkSourceInfo(sender nexus.CrossChainAddress, msg map[string]interface{}) error {
-	chain, ok := msg[sourceChain]
-	if ok && !sender.Chain.Name.Equals(nexus.ChainName(fmt.Sprint(chain))) {
-		return fmt.Errorf("source chain does not match expected")
+	chainI, ok := msg[sourceChain]
```

Comment:
what is `chainI`, is that on purpose?

Reply:
`msg` has type `map[string]interface{}`. `chainI` will have the type `interface{}`, `chain` will have `string`. Any other naming suggestions?

Reply:
why not keep chain? It's going to get shadowed by the type cast version

Reply:
ah, yes, forgot you re-declare it if original type was interface{}